### PR TITLE
Implement delimiter and eol events

### LINKED ISSUE #252: Getting auto detected delimiter and eol
`csvtojson` can infer delimiter, eol, and headers. The inferred headers are returned on `header` or `json` events. But is there a way to get back the delimiter and end of line sequence that were detected?

### PR DESCRIPTION
**Pull request by gabegorelick**
Closes #252.

### CODE REVIEW

File: test/data/dataNoTrimCRLF [hunk 1]
```diff
@@ -1,3 +1,3 @@
-name,age
-joe,20
+name,age
+joe,20
```
Comment (gabegorelick) [inline test/data/dataNoTrimCRLF]: This file didn't actually have CRLF line endings. I assume that was accidental?

File: package.json [hunk 2]
```diff
@@ -278,7 +278,8 @@
     "load-grunt-tasks": "^3.4.0",
     "minimist": "^1.2.0",
     "mocha": "^2.4.5",
-    "request": "^2.81.0"
+    "request": "^2.81.0",
+    "sinon": "^5.0.3"
```
Comment (gabegorelick) [inline package.json]: Using sinon spies to verify the events are only emitted once. Let me know if you prefer something else.

File: test/testCSVConverter.js [hunk 3]
```diff
@@ -410,6 +415,49 @@ describe("CSV Converter", function () {
...
+      assert.equal(eol, "\n");
+    });
+    st.on("eol", eolCallback);
+    st.on("end_parsed", function () {
+      assert.equal(eolCallback.callCount, 1, 'should emit eol event once');
+      done();
+    })
+  });
+
+  it("should not emit eol event when eol is specified", function (done) {
```
Comment (gabegorelick) [inline test/testCSVConverter.js]: Not sure what the best behavior is here. Does it make sense to always emit it?

File: test/data/dataNoTrimCRLF (hunk 1 above)
Comment (Keyang) [reply test/data/dataNoTrimCRLF]: hmm... This was weird. But will change to CRLF...

File: package.json (hunk 2 above)
Comment (Keyang) [reply package.json]: sinon is good and agree to use it
`v2.0.0` will drop to any version < 4. So I will temporarily comment out  the multiple call check and add it back in `v2.0.0`

File: test/testCSVConverter.js (hunk 3 above)
Comment (Keyang) [reply test/testCSVConverter.js]: I think we need emit it if user subscribes to the event.

**Comment by Keyang** [comment]
Thanks @gabegorelick. I will merge it in next days and make some changes in my comments above.  Then I will rollout a version `1.1.10` .

**Pull request merged**